calculations.py

```diff
@@ -0,0 +1,118 @@
+def generate_unique_items_list(data):
+    items_id_column_index = 0
+    items_id_list = []
+
+    for line in data:
+        if not line[items_id_column_index] in items_id_list:
+            items_id_list.append(line[items_id_column_index])
+
+    return items_id_list
+
+
+def count_unique_items(data):
+    return len(generate_unique_items_list(data))
+
+
+def generate_unique_shops_list(data):
+    shop_id_column_index = 1
+    shop_id_list = []
+
+    for line in data:
+        if not line[shop_id_column_index] in shop_id_list:
+            shop_id_list.append(line[shop_id_column_index])
+
+    return shop_id_list
+
+
+def count_unique_shops(data):
+    return len(generate_unique_shops_list(data))
+
+
+def generate_user_statistic(data):
+    user_column_index = 3
+    user_statistic = {}
+
+    for line in data:
+        if not line[user_column_index] in user_statistic.keys():
+            user_statistic[line[user_column_index]] = 1
+        else:
+            user_statistic[line[user_column_index]] += 1
```

Comment:
`defaultdict` or `Counter` would be much easier to use https://docs.python.org/3.8/library/collections.html